Edits to pgd_bench content (#4789)
EnterpriseDB/docs: product_docs/docs/pgd/5/reference/testingandtuning.mdx

indexdepth: 2
---

EDB Postgres Distributed has tools that help with testing and tuning your PGD clusters. For background, see [Testing and tuning](../testingandtuning).

> ## pgd_bench

Reviewer: ```suggestion
## `pgd_bench`
```

This is a reference page for a command and follows the reference styling.

### Synopsis

> A benchmarking tool for PGD-enhanced PostgreSQL.

Reviewer 2: I'd like this to say EDB Postgres Distributed deployments

Reviewer: ```suggestion
A benchmarking tool for EDB Postgres Distributed deployments.
```

```shell
pgd_bench [OPTION]... [DBNAME] [DBNAME2]
```

`DBNAME` can be a conninfo string of the format:
  `"host=10.1.1.2 user=postgres dbname=master"`

See [pgd_bench in Testing and tuning](../testingandtuning#pgd_bench) for examples
of pgd_bench options and usage.

### Options

> pgd_bench-specific options include the following.

Reviewer: ```suggestion
The pgd_bench command is implemented as a wrapper around the pgbench command. This means that it shares many of the same options and created tables named `pgbench` as it performs its testing.

Options that are specific to pgd_bench include the following.
```

#### Setting mode

`-m` or `--mode`

The mode can be set to `regular`, `camo`, or `failover`. The default is `regular`.

* `regular` &mdash; Only a single node is needed to run pgd_bench.
* `camo` &mdash; A second node must be specified to act as the CAMO partner. (CAMO must be set up.)
* `failover` &mdash; A second node must be specified to act as the failover.

When using `-m failover`, an additional option `--retry` is available. This option
instructs pgd_bench to retry transactions when there's a failover. The `--retry` 

> option is enabled with `-m camo`.

Reviewer 2: I think automatically should be added back

Reviewer: ```suggestion
option is automatically enabled when `-m camo` is used.
```

#### Setting GUC variables

 `-o` or `--set-option`
 
This option is followed by `NAME=VALUE` entries, which are applied using the 

> PostgreSQL [`SET`](https://www.postgresql.org/docs/current/sql-set.html) command on each server that pgd_bench connects to, and only those servers. 

Reviewer 2: I'd change to Postgres

> The other options are identical to the PostgreSQL pgd_bench command. For

Reviewer 2: I'd change all references to PostgreSQL to Postgres - these options are available regardless of whether EDB Postgres Distributed is using PostgreSQL, EDB Postgres Extended Server or EDB Postgres Advanced Server

Reviewer: ```suggestion
The other options are identical to the Postgres pgbench command. For
```

> The other options are identical to the PostgreSQL pgd_bench command. For

Reviewer 2: I think this is supposed to say pg_bench - I think the point is the the pgd_bench commands are the same as the pg_bench commands

Reviewer: Yes it is, fixed above.

details, see the PostgreSQL 

> [pgd_bench](https://www.postgresql.org/docs/current/pgbench.html) documentation.

Reviewer: ```suggestion
[pgbench](https://www.postgresql.org/docs/current/pgbench.html) documentation.
```

The complete list of options (pgd_bench and pgbench) follow.

> 

Reviewer 2: we're using inconsistent references to pgbench and pgd_bench below - let's pick one or say it a different way - e.g. drop existing benchmark tables, create the tables used by the standard benchmarking scenario

maybe check with devs if pgbench_accounts is not changed to pgd - then i think changing the above will help avoid the inconsistency - we may want to add a note (if we don't already have one) that pgd_bench is wrapper around pgbench and all table names will include pgbench

Reviewer: It is not changed. Fixed.

#### Initialization options
-   `-i, --initialize`  &mdash; Invoke initialization mode.
-   `-I, --init-steps=[dtgGvpf]+` (default `"dtgvp"`)  &mdash; Run selected initialization steps.

>     -   `d`  &mdash; Drop any existing pgd_bench tables.

Reviewer: ```suggestion
    -   `d`  &mdash; Drop any existing pgbench tables.
```

>     -   `t`  &mdash; Create the tables used by the standard pgd_bench scenario.

Reviewer: ```suggestion
    -   `t`  &mdash; Create the tables used by the standard pgbench scenario.
```
